### ISSUE TITLE
New content of edited replies should include the reply itself

### ISSUE THREAD

**Issue by schmop** [vector-im/riot-web#13811]
### Description

When sending a reply to a text message, the content looks like this:
```
{
  content: {
    body: " > <[Matrix ID]> [Replied Preview] \n\n [Reply]",
    formatted_body: "<mx-reply>[...]</mx-reply>[Reply]",
    format: "org.matrix.custom.html"
  }
}
```
But the new content does not include these fallback reply representations:
```
{
  "m.new_content": {
    format: "org.matrix.custom.html",
    body: "[NewReply]",
    formatted_body: "[NewReply]"
  },
}
```

Especially the matrix-js-sdk overrides the old content of the event on "m.replace" events.
Therefore the fallback reply will be omitted, but i think it shouldn't.

### Steps to reproduce

- Write a textmessage
- Reply to textmessage
- Edit the reply to the textmessage

### Expectation

```
{
  "m.new_content": {
    format: "org.matrix.custom.html",
    body: "> <[MatrixId]> \n\n [NewReply]",
    formatted_body: "<mx-reply>[...]</mx-reply>[NewReply]"
  },
}
```

**Comment by t3chguy** [comment]
The fallback is only there for backwards compatibility with clients which don't understand `m.new_content` anyway.

**Comment by schmop** [comment]
Im especially not talking about the fallback representation of editing "m.replace" events, but for reply fallbacks.

**Comment by t3chguy** [comment]
Yes, I know, they each have their own fallback.
I think the decision here was if a client doesn't understand replies then it won't understand edits either and thus it will never see `m.new_content`

**Comment by schmop** [comment]
Yeah, but I don't think it's very consistent to include the fallback everywhere but in this very case.
We just got the case, where riot-web sent events to a third-party client, that relied on these fallbacks.

We got it fixed on our side now, but I fear that other client combinations might get into similar problems

**Comment by MadLittleMods** [comment]
It's also useful to include the new reply fallback in the edited message when you edit what message you're replying to, https://github.com/vector-im/element-web/issues/10391
